### PR TITLE
Spring animation, over scroll and android bug fix

### PR DESCRIPTION
**Pull request by dcoulter45**
- Added an option to use spring animation after a gesture release for nicer user experience
- Added an option for over scroll so that users can pull the drawer lower than it's last snapshot 
- Fixed a bug on android where getNode would at times have a missing function

Spring animation is based off @tankers746 pull request, so thank you to him for that.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4350,6 +4350,7 @@
             "babel-polyfill": "6.26.0",
```
Comment (rgommezz) [inline package-lock.json:4350]: Could we please revert this change?

File: src/index.tsx [hunk 2]
```diff
@@ -129,6 +140,12 @@ interface TimingParams {
   position: Animated.Value<number>;
   finished: Animated.Value<number>;
   frameTime: Animated.Value<number>;
+  velocity: Animated.Node<number>;
+}
+
+export enum AnimationType {
+  Timing,
+  Spring,
 }
```
Comment (rgommezz) [inline src/index.tsx:149]: We can delete `AnimationType`, it's not used anywhere

```suggestion
```

File: src/index.tsx [hunk 3]
```diff
@@ -160,12 +177,9 @@ type CommonProps = {
...
   animatedPosition?: Animated.Value<number>;
   /**
-   * Configuration for the timing reanimated function
+   * Configuration for the timing or spring reanimated function
    */
-  animationConfig?: {
-    duration?: number;
-    easing?: Animated.EasingFunction;
-  };
+  animationConfig?: Animated.SpringConfig & Animated.TimingConfig;
```
Comment (rgommezz) [inline src/index.tsx:182]: `animationConfig` prop definition can be removed as well from `CommonProps`, it's overridden and documented further down in lines 214 and 222

File: src/index.tsx [hunk 4]
```diff
@@ -436,43 +490,54 @@ export class ScrollBottomSheet<T extends any> extends Component<Props<T>> {
...
+          DEFAULT_EASING,
+        toValue: new Value(0),
       };
 
-      const config = {
+      const springConfig = {
+        ...((props.animationType === 'spring' && props.animationConfig) || DEFAULT_SPRING_PARAMS),
         toValue: new Value(0),
-        ...animationParams,
       };
```
Comment (rgommezz) [inline src/index.tsx:521]: We need to merge the provided spring config with the defaults, just in case user only wants to customise some of the values.

```ts
      const springConfig = {
		...DEFAULT_SPRING_PARAMS,
        ...((props.animationType === 'spring' && props.animationConfig) ||
          DEFAULT_SPRING_PARAMS),
        toValue: new Value(0),
      };
```

**Review by rgommezz — COMMENTED** [review]
Thanks for the changes, some last polishing and this is good to go!

Could you please revert the changes in the `package-lock.json` as well?

**Comment by dcoulter45** [comment]
> Thanks for the changes, some last polishing and this is good to go!
> 
> Could you please revert the changes in the `package-lock.json` as well?

No problem, made those last changes. Hopefully it all looks good.

File: src/index.tsx [hunk 5]
```diff
@@ -436,43 +484,55 @@ export class ScrollBottomSheet<T extends any> extends Component<Props<T>> {
...
+        easing:
+          (props.animationType === 'timing' && props.animationConfig?.easing) ||
+          DEFAULT_EASING,
+        toValue: new Value(0),
       };
 
-      const config = {
+      const springConfig = {
+        ...DEFAULT_SPRING_PARAMS,
+        ...(props.animationType === 'spring' && props.animationConfig),
```
Comment (rgommezz) [inline src/index.tsx:514]: That could throw if `animationType === 'spring'` and `animationConfig` is not defined. We need to default it to an empty object.

```suggestion
        ...((props.animationType === 'spring' && props.animationConfig) || {}),
```

**Review by rgommezz — APPROVED** [review]
LGTM! 🎉
